connect/runtime/src/main/java/org/apache/kafka/connect/util/PluginVersionUtils.java

```diff
@@ -0,0 +1,186 @@
+package org.apache.kafka.connect.util;
+
+import org.apache.kafka.common.config.ConfigDef;
+import org.apache.kafka.common.config.ConfigException;
+import org.apache.kafka.connect.connector.Connector;
+import org.apache.kafka.connect.runtime.ConnectorConfig;
+import org.apache.kafka.connect.runtime.isolation.PluginDesc;
+import org.apache.kafka.connect.runtime.isolation.Plugins;
+import org.apache.kafka.connect.runtime.rest.entities.ConnectorType;
+import org.apache.maven.artifact.versioning.InvalidVersionSpecificationException;
+import org.apache.maven.artifact.versioning.VersionRange;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+public class PluginVersionUtils {
+
+    private static Plugins plugins = null;
```

Comment:
You need to find a way to make this non-static; There are routinely multiple Plugins instances in use within the same JVM, which would immediately collide on this field.

Reply:
Can you elaborate on how this would collide? IIUC this static variable should still be scoped to the `PluginVersionUtils` class and gets initialized explicitly after the Plugins instance in created in `AbstractConnectCli`. Any other instance should not affect or alter this. I did not find any issues with my testing, but LMK if I should look from anything.

Reply:
The EmbeddedConnectCluster creates one worker, which instantiates `new Plugins`, and then sets the PluginVersionUtils to that instance.

The EmbeddedConnectCluster creates another worker, which instantiates `new Plugins`, and then overwrites the field in PluginVersionUtils.

The first worker and the second worker use all of the same Recommender instances and the same Plugins instances, even though they could be started with completely different plugin.path.

Reply:
Thanks for catching this. I created a new `PluginsRecommender` class that is instantiated from a `plugins` instance so that we can avoid this problem. This also partially addresses your previous comment https://github.com/apache/kafka/pull/16984#discussion_r1868500756 and reduces most of the new exposed methods from plugins instance.